### PR TITLE
Refactor plan package

### PR DESCRIPTION
**Pull request by barrettj12**
#152 exported the `plan` package so that external Go projects could use the types defined there to easy write Pebble layers. However, this package also dragged in a bunch of extra utilities, causing compilation problems on non-Unix. This PR does the following:

- Create a new package `internal/planutil`
- Move the following methods (and associated constants / helper methods) from `plan` into `internal/planutil`:
  - CombineLayers
  - StartOrder
  - StopOrder
  - ParseLayer
  - ReadLayersDir
  - ReadDir
- Move the `plan` tests into `internal/planutil`
- Update imports for things which moved from `plan` to `internal/planutil`.

Now, importing the `github.com/canonical/pebble/plan` package will give just the struct definitions, which should fix the compilation issues.

### CODE REVIEW

**Review by tlm — COMMENTED** [review]
lgtm @barrettj12

File: layer/layer.go [hunk 1]
```diff
@@ -0,0 +1,355 @@
+package layer
```
Comment (hpidcock) [inline layer/layer.go:1]: Missing license.

Comment (barrettj12) [reply layer/layer.go:15]: Good catch, a few other files were missing licenses too so I added these.

File: internal/overlord/servstate/manager.go [hunk 2]
```diff
@@ -118,7 +133,7 @@ func (m *ServiceManager) Plan() (*plan.Plan, error) {
 // AppendLayer appends the given layer to the plan's layers and updates the
 // layer.Order field to the new order. If a layer with layer.Label already
 // exists, return an error of type *LabelExists.
-func (m *ServiceManager) AppendLayer(layer *plan.Layer) error {
+func (m *ServiceManager) AppendLayer(layer *layer.Layer) error {
```
Comment (benhoyt) [inline internal/overlord/servstate/manager.go:136]: Hmm, yeah, `layer` is such a good (and common) variable name. This shadowing is allowed in Go, but it's a bit odd and confusing: "in this piece of code, is `layer` the package or a variable?"

Perhaps would should call the package `layers` instead? Or even `types`? (`types.Layer` reads pretty well.)

File: internal/overlord/servstate/manager.go [hunk 3]
```diff
@@ -177,36 +192,36 @@ func findLayer(layers []*plan.Layer, label string) (int, *plan.Layer) {
 // CombineLayer combines the given layer with an existing layer that has the
 // same label. If no existing layer has the label, append a new one. In either
 // case, update the layer.Order field to the new order.
-func (m *ServiceManager) CombineLayer(layer *plan.Layer) error {
+func (m *ServiceManager) CombineLayer(l *layer.Layer) error {
```
Comment (benhoyt) [inline internal/overlord/servstate/manager.go:195]: Sort of proves the point above about the package name.

**Review by tlm — APPROVED** [review]

File: internal/planutil/plan.go [hunk 4]
```diff
@@ -0,0 +1,506 @@
+// Copyright (c) 2021 Canonical Ltd
```
Comment (benhoyt) [inline internal/planutil/plan.go:1]: This file should probably be called `planutil.go` now.

File: internal/planutil/plan_test.go [hunk 5]
```diff
@@ -13,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package plan_test
+package planutil_test
```
Comment (benhoyt) [inline internal/planutil/plan_test.go:15]: And this file `planutil_test.go`.

**Review by benhoyt — APPROVED** [review]
Looks good to me! Let's just change the `planutil` filenames to `planutil*.go`.

**Comment by niemeyer** [comment]
Actually, given go.mod supposedly this should not be a problem since no code is supposed to break upfront without an update, so I going to revert it so that we can stack the upcoming PRs without this change.

**Pull request merged**